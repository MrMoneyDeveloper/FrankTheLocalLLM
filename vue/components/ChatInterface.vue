<template>
  <div>
    <input v-model="message" placeholder="Say hi" />
    <button @click="send">Send</button>
    <p v-if="reply">{{ reply }}</p>
  </div>
</template>
<script setup>
import { ref } from 'vue'

const message = ref('')
const reply = ref('')


async function send() {
  reply.value = ''
  const resp = await fetch('/api/qa/stream', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ question: message.value })
  })
  const reader = resp.body.getReader()
  const decoder = new TextDecoder('utf-8')
  while (true) {
    const { value, done } = await reader.read()
    if (done) break
    reply.value += decoder.decode(value)
  }
}
</script>
