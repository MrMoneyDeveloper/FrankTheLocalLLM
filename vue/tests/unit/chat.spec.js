import { mount } from '@vue/test-utils'
import ChatInterface from '../../components/ChatInterface.vue'
import { describe, it, expect, vi } from 'vitest'
import flushPromises from 'flush-promises'

global.fetch = vi.fn(() => Promise.resolve({
  body: {
    getReader () {
      let done = false
      return {
        async read () {
          if (done) return { done: true }
          done = true
          return { done: false, value: new TextEncoder().encode('hi there') }
        }
      }
    }
  }
}))

describe('ChatInterface', () => {
  it('sends message and shows reply', async () => {
    const wrapper = mount(ChatInterface)
    await wrapper.find('input').setValue('hello')
    await wrapper.vm.send()
    await flushPromises()
    expect(fetch).toHaveBeenCalled()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
