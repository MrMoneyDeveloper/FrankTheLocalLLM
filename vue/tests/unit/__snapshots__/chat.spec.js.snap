// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatInterface > sends message and shows reply 1`] = `
"<div><input placeholder="Say hi"><button>Send</button>
  <p>hi there</p>
</div>"
`;
